pkg/util/mount/linux.go

```diff
@@ -47,10 +50,44 @@ func (mounter *mounter) Unmount(target string, flags int) error {
 	return syscall.Unmount(target, flags)
 }
 
+// safe_format_and_mount is a utility script on GCE VMs that probes a persistent disk, and if
+// necessary formats it before mounting it.
+// This eliminates the necesisty to format a PD before it is used with a Pod on GCE.
+// TODO: port this script into Go and use it for all Linux platforms
+type gceSafeFormatAndMount struct {
+	procMountLister
+	runner exec.Interface
+}
+
+// uses /usr/share/google/safe_format_and_mount to optionally mount, and format a disk
+func (mounter *gceSafeFormatAndMount) Mount(source string, target string, fstype string, flags uintptr, data string) error {
```

Comment:
You can "subclass" it in GCE code without disassembling it.

http://play.golang.org/p/x3b_x8gKod

If you want to commit this, I will make the transform, since I am so delinquent.